jobsdb/prebackup/prebackup_test.go

```diff
@@ -0,0 +1,194 @@
+package prebackup
+
+import (
+	"context"
+	"database/sql"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"testing"
+
+	"github.com/ory/dockertest"
+	"github.com/rudderlabs/rudder-server/testhelper"
+	"github.com/rudderlabs/rudder-server/testhelper/destination"
+)
+
+var postgresql *destination.PostgresResource
+
+func Test_DropSourceIdsHandler(t *testing.T) {
+	handler := DropSourceIds(func() []string { return []string{"one", "two"} })
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	txn, err := postgresql.DB.Begin()
+	if err != nil {
+		t.Fatalf("Failed to create transaction: %s", err)
+	}
+	const (
+		jobsTable      = "jobs"
+		jobStatusTable = "job_status"
+	)
+
+	// Given I have jobs and job status tables with 3 events
+	createTables(t, txn, jobsTable, jobStatusTable)
+	addJob(t, txn, jobsTable, jobStatusTable, "one", "succeeded")
+	addJob(t, txn, jobsTable, jobStatusTable, "two", "failed")
+	addJob(t, txn, jobsTable, jobStatusTable, "three", "aborted")
+
+	if rows := countRows(t, txn, jobsTable); rows != 3 {
+		t.Fatalf("Expected 3 rows in jobs table, got %d", rows)
+	}
+	if rows := countRows(t, txn, jobStatusTable); rows != 3 {
+		t.Fatalf("Expected 3 rows in job status table, got %d", rows)
+	}
+
+	// when I execute the dropSourceIds handler for the 2 source_ids
+	if err = handler.Handle(ctx, txn, jobsTable, jobStatusTable); err != nil {
+		t.Fatalf("Failed to handle: %s", err)
+	}
+
+	// then I end up with one event on each table
+	if rows := countRows(t, txn, jobsTable); rows != 1 {
+		t.Fatalf("Expected 1 row in jobs table, got %d", rows)
+	}
+	if rows := countRows(t, txn, jobStatusTable); rows != 1 {
+		t.Fatalf("Expected 1 row in job status table, got %d", rows)
+	}
+
+}
+func Test_DropSourceIdsHandler_Without_SourceIds(t *testing.T) {
+	handler := DropSourceIds(func() []string { return []string{} })
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	var txn *sql.Tx
+
+	// when I execute the dropSourceIds handler for an empty source ids list, a canceled context and nil transaction
+	err := handler.Handle(ctx, txn, "jobs", "job_status")
+
+	// then handler doesn't do anything and no error is returned
+	if err != nil {
```

Comment:
[nitpick] Could use `require.NoError(t, err)` to make it shorter, ofc totally up to you.

Reply:
ok